whylog/constraints/verifier.py

```diff
@@ -0,0 +1,110 @@
+from whylog.constraints import IdenticalIntervals, TimeConstraint
+from whylog.exceptions import WhylogError
+from whylog.front import FrontInput
+
+
+class ConstraintManager(object):
+    @classmethod
+    def get_constraint_by_type(cls, constraint_data):
+        constraints = {
+            'identical': IdenticalIntervals({}),
+            'time': TimeConstraint({})
+            # register your constraint here
+        }
+        if constraint_data['name'] in constraints:
+            return constraints[constraint_data['name']]
+        raise WhylogError("No such constraint (%s) registered" % constraint_data['name'])
+
+
+class Verifier(object):
+    @classmethod
+    def _front_input_from_clue(cls, clue):
+        return FrontInput(clue.line_offset, clue.line_prefix_content, clue.line_source)
+
+    @classmethod
+    def _verify_constraint(cls, combination, effect, constraint):
+        constraint_verifier = ConstraintManager.get_constraint_by_type(constraint)
+        groups = []
+        for group_info in constraint['clues_groups']:
+            parser_num, group_num = group_info[0], group_info[1]
+            if parser_num == 0:
+                groups.append(effect.regex_parameters[group_num - 1])
+            else:
+                groups.append(combination[parser_num - 1].regex_parameters[group_num - 1])
```

Comment:
Now, You can get index out of range error here, when combination length is less than number of causes parsers in group.